esmvalcore/_config/_config.py

```diff
@@ -14,14 +14,24 @@
 
 CFG = {}
 
+try:
+    from importlib.resources import files as importlib_files
+except ImportError:
+    from importlib_resources import files as importlib_files
+
 
 @lru_cache
 def _get_project_mappings(project, dataset):
-    DEFAULT_PATH = (Path(__file__).parents[0] / project
-                    / f"{dataset}-mappings.yml")
-    mapping_path = CFG.get(project, {}).get("mapping_path", DEFAULT_PATH)
```

Comment:
Deleting the possibility to have a mapping path defined in the config-developer file means keeping users unable to re-define the mapping when they change the way their model's outputs
(but maybe I misunderstand which is the actual value of SEARCH_PATHS